sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/AppView.groovy

```diff
@@ -74,6 +78,16 @@ class AppView extends VerticalLayout {
         }
         projectSampleToggle.addClickListener(switchToSampleView, ToggleButton.State.ONE)
         projectSampleToggle.addClickListener(switchToProjectView, ToggleButton.State.TWO)
+
+        projectOverviewView.onProjectDoubleClick({
+            sampleOverviewView.getController().getSamplesFor(it.code)
+            toggle()
+        })
+    }
+
+    private void toggle(){
+        showProjectView(!projectOverviewView.isVisible())
+        showSampleView(!sampleOverviewView.isVisible())
```

Comment:
The ToggleButton provides this functionality already. It can be reused by making the `toggleWithEvent` method public. If done this way, the public `toggle` method has to be called so that the button is in the right state.